src/main/java/gil/server/Router.java

```diff
@@ -0,0 +1,38 @@
+package gil.server;
+
+import java.io.UnsupportedEncodingException;
+import java.util.HashMap;
+import java.util.function.Function;
+
+public class Router {
+    private HashMap<String, Function<Response, Response>> routes;
+
+    public Router() {
+        routes = new HashMap<>();
+        routes.put("/", (Response response) -> {
```

Comment:
This route / response registration functionality in the constructor will inevitably grow given the server's future requirements. This seems like a good opportunity to refactor this responsibility to either a new method in the router itself or some sort of other abstraction.